### PR TITLE
Mon 12982 downtime by

### PR DESCRIPTION
**Pull request by bouda1**
## Description

Downtimes are stored with hosts and services id now.

REFS: MON-12982

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

### CODE REVIEW

File: engine/src/commands/commands.cc [hunk 1]
```diff
@@ -855,132 +855,129 @@ int cmd_remove_acknowledgement(int cmd, char* args) {
...
   time_t end_time{0};
   bool fixed{false};
   uint64_t triggered_by{0};
   unsigned long duration{0};
-  char* author{nullptr};
-  char* comment_data{nullptr};
   uint64_t downtime_id{0};
   servicegroup_map::const_iterator sg_it;
 
+  std::vector<absl::string_view> a{absl::StrSplit(args, ';')};
```
Comment (jean-christophe81) [inline engine/src/commands/commands.cc:869]: Is auto more efficient to avoid conversion to vector?

File: engine/src/commands/commands.cc [hunk 2]
```diff
@@ -1146,57 +1146,75 @@ int cmd_delete_downtime(int cmd, char* args) {
  *  @param[in] args  Command arguments.
  */
 int cmd_delete_downtime_full(int cmd, char* args) {
-  char* temp_ptr(nullptr);
+  log_v2::functions()->trace("cmd_delete_downtime_full() args = {}", args);
   downtime_finder::criteria_set criterias;
 
+  std::vector<absl::string_view> a{absl::StrSplit(args, ';')};
```
Comment (jean-christophe81) [inline engine/src/commands/commands.cc:1152]: idem

File: engine/src/comment.cc [hunk 3]
```diff
@@ -114,16 +119,20 @@ void comment::delete_host_comments(uint64_t host_id) {
           it->second->get_comment_data().c_str(), it->second->get_persistent(),
           it->second->get_source(), it->second->get_expires(),
           it->second->get_expire_time(), it->first);
-      it = comments.erase(it);
+      comments.erase(it++);
```
Comment (jean-christophe81) [inline engine/src/comment.cc:122]: it = comments.erase(it)

File: engine/src/comment.cc [hunk 4]
```diff
@@ -134,19 +143,22 @@ void comment::delete_service_comments(uint64_t host_id, uint64_t service_id) {
...
```
Comment (jean-christophe81) [inline engine/src/comment.cc:146]: idem

File: engine/src/comment.cc [hunk 5]
```diff
@@ -157,21 +169,23 @@ void comment::delete_host_acknowledgement_comments(engine::host* hst) {
...
```
Comment (jean-christophe81) [inline engine/src/comment.cc:172]: idem

File: engine/src/comment.cc (hunk 3 above)
Comment (bouda1) [reply engine/src/comment.cc:122]: this the recommanded way by absl. Their method is a void method.

Comment (bouda1) [reply engine/src/comment.cc:122]: this would not compile

File: engine/src/comment.cc (hunk 4 above)
Comment (bouda1) [reply engine/src/comment.cc:146]: this would not compile

File: engine/src/comment.cc (hunk 5 above)
Comment (bouda1) [reply engine/src/comment.cc:172]: would not compile

File: engine/src/comment.cc [hunk 6]
```diff
@@ -183,9 +197,9 @@ void comment::delete_service_acknowledgement_comments(::service* svc) {
...
```
[the remark this answers is not in the document]
Comment (bouda1) [reply engine/src/comment.cc:200]: would not compile

**Pull request merged**